### Manipulación de datos con Numpy

**1.1 Setup del Entorno y Configuración**

- setup.md

**1.2Álgebra lineal aplicada**

In [5]:
#importando las librerias correspondientes

import numpy as np
import pandas as pd


In [6]:
#cargando los datos de csv y creando los arrays de las cantidades y de los precios
#y accediendo a ellos por medio de un ciclo for
datos = np.genfromtxt("../data/ventas_techventas.csv", encoding="latin-1", delimiter=",", skip_header=1)
cantidad = datos[:,7]
precio = datos[:,8]

print("Cantidad\tPrecio")

for i in range(len(cantidad)):
    print(f"{cantidad[i]}\t\t{precio[i]}")


Cantidad	Precio
2.0		1200.0
15.0		25.0
5.0		350.0
10.0		85.0
3.0		1200.0
20.0		95.0
2.0		350.0
8.0		120.0
12.0		95.0
1.0		1200.0
4.0		350.0
30.0		25.0
5.0		85.0
3.0		120.0
25.0		95.0
2.0		1200.0
6.0		350.0
18.0		95.0
12.0		25.0
2.0		1200.0
4.0		120.0
30.0		95.0
8.0		85.0
3.0		350.0
4.0		1200.0
20.0		25.0
15.0		95.0
6.0		120.0
1.0		1200.0
10.0		85.0
3.0		350.0
22.0		95.0
5.0		1200.0
25.0		25.0
2.0		120.0
4.0		350.0
12.0		85.0
18.0		95.0
2.0		350.0
3.0		1200.0
15.0		25.0
5.0		120.0
2.0		350.0
3.0		1200.0
20.0		95.0
6.0		85.0
18.0		25.0
-1.0		1200.0
4.0		120.0
25.0		95.0
3.0		350.0
2.0		1200.0
20.0		25.0
35.0		95.0
5.0		350.0
2.0		1200.0
7.0		120.0
9.0		85.0
20.0		95.0
4.0		350.0


In [7]:
#Realizando el calculo del producto punto tomando los valores de cantidad y precio 
# y la sumatoria de total de las columnas
ingreso_bruto = cantidad @ precio
print("Comparando valores (Sin limpiar los datos)"),
print("\nCantidad total de productos vendidos: ",cantidad.sum())
print("\nSumatoria del precio Total de todos los productos: ",precio.sum())
print("\nIngreso Bruto: ",ingreso_bruto)

Comparando valores (Sin limpiar los datos)

Cantidad total de productos vendidos:  586.0

Sumatoria del precio Total de todos los productos:  22695.0

Ingreso Bruto:  88205.0


**1.3 Estadísticas con Numpy**

In [8]:
media = np.mean(precio)
mediana = np.median(precio)
desviacion_estandar = np.std(precio)
percent = np.percentile(precio,[25,75])
print("Media de los precios: ", media)
print("\nMediana de los precios: ", mediana)
print("\nDesviación estándar de los precios: ", round(desviacion_estandar,2))
print("\nPercentiles 25 y 75: ", percent)

Media de los precios:  378.25

Mediana de los precios:  120.0

Desviación estándar de los precios:  445.06

Percentiles 25 y 75:  [ 92.5 350. ]


**Análisis**

- En base a las operaciones realizadas se puede observar que el promedio del precio de la cantidad de productos vendidos corresponde a 378.25 dólares durante los ultimos 6 meses. La mediana de las ventas corresponde a 120.0 dolares, lo que se podria interpretar en que al menos la mitad de las ventas oscila alrededor de este valor ($120). En comparación con el promedio, la desviación estándar es de 445.06 dolares, de lo que se podría decir que el valor de las ventas varia del promedio con una cantidad de $66.81 (diferencia de la desviación estándar y el promedio) y, los percentiles que presentan los valores de 92.5 y 350, lo que significa que el 25% de las ventas esta debajo de $92.5 y el 75% esta por encima de $350.

**1.4 Reflexión CRISP-DM**

- Las fases que considero mas importantes respecto a la metodologia son las 1 y 2 que son respecto a la comprención del negocio y la comprension de los datos; en base a la experiencia que tuve realizando esta parte del proyecto solamente cuando se comprende el problema, las necesidades del negocio y tambien los objetivos del mismo se puede proceder a analizar los datos para obtener la informacion necesaria para llegar a concluciones y ya saber proceder con los demas pasos requeridos, hasta llegar a las conclusiones mas acertadas. Al hacer este ejercicio pude darme cuenta que de nada sirve manipular los datos si no se entiende el objetivo para el que se hace. Creo que eso permite que al final se puedan tomar las desiciones correctas.